sklearn/conftest.py

```diff
@@ -37,6 +38,17 @@
     "fetch_rcv1_fxt": fetch_rcv1,
 }
 
+_SKIP32_MARK = pytest.mark.skipif(
+    environ.get("SKLEARN_SKIP_FLOAT32", "1") != "0",
+    reason="Set SKLEARN_SKIP_FLOAT32=0 to run float32 dtype tests",
+)
+
+
+# Global fixture
+@pytest.fixture(params=[pytest.param(np.float32, marks=_SKIP32_MARK), np.float64])
+def dtype(request):
```

Comment:
We use "dtype" as a `pytest.mark.parameterize" in many places where we actually want to test for all dtypes. Maybe we need to give this a more semantic name:

```suggestion
def global_dtype(request):
```

I am open for other suggestions.